Src/FluentAssertions/Equivalency/GenericEnumerableEquivalencyStep.cs

```diff
@@ -35,14 +34,16 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
         {
             Type expectedType = config.GetExpectationType(context);
 
-            var interfaceTypes = GetIEnumerableInterfaces(expectedType)
-                .Select(type => "IEnumerable<" + type.GetGenericArguments().Single() + ">")
-                .ToList();
+            var interfaceTypes = GetIEnumerableInterfaces(expectedType);
 
-            AssertionScope.Current
-                .ForCondition(interfaceTypes.Count == 1)
-                .FailWith("{context:Expectation} implements {0}, so cannot determine which one " +
-                          "to use for asserting the equivalency of the collection. ", interfaceTypes);
+            if (interfaceTypes.Length != 1)
```

Comment:
What measurable difference does the added conditional make?

Reply:
About 0.3% overall, 8% of this method (excl. the `HandleImpl` invoke).

Reply:
Changed to use the new overload.